### PR TITLE
Added unit tests for nuget package and added new build and test workflow

### PR DESCRIPTION
**Pull request by eboaden**

### CODE REVIEW

File: packages/nuget/Features.Tests/Providers/NamespaceProviderTests.cs [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+namespace Features.Tests.Providers
+{
+    public class NamespaceProviderTests
+    {
+        [Fact]
+        public void GetNamespaceFromNamespaceFile()
+        {
+            File.WriteAllText("namespace.txt", "file-ns");
+            var ns = new NamespaceProvider(namespacePath: "namespace.txt", ns: "demo-app").GetNamespace();
+            Assert.Equal("file-ns", ns);
```
Comment (JoeSainsburys) [inline packages/nuget/Features.Tests/Providers/NamespaceProviderTests.cs:15]: make const

File: packages/nuget/Features.Tests/FeaturesClientTests.cs [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+        }
+
+        [Fact]
+        public void CheckFeatureIsNotActiveWhenFeatureNameIsProvided()
+        {
+            MockClientCall(_mockNamespaceProvider, _mockKubernetesClient, "test-ns", "test-feature", false);
+
+            var featuresClient = new FeaturesClient(namespaceProvider: _mockNamespaceProvider.Object,
+                clientProvider: _mockKubernetesClientProvider.Object);
+            var featureResponse = featuresClient.CheckFeatureIsActive("test-feature");
```
Comment (JoeSainsburys) [inline packages/nuget/Features.Tests/FeaturesClientTests.cs:46]: make test-feature a const

File: packages/nuget/Features.Tests/FeaturesClientTests.cs [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+        
+        [Fact]
+        public void CheckFeatureIsActiveWhenFeatureNameIsProvided()
+        {
+            
+            MockClientCall(_mockNamespaceProvider, _mockKubernetesClient, "test-ns", "test-feature", true);
+
+            var featuresClient = new FeaturesClient(namespaceProvider: _mockNamespaceProvider.Object,
+                clientProvider: _mockKubernetesClientProvider.Object);
+            var featureResponse = featuresClient.CheckFeatureIsActive("test-feature");
```
Comment (JoeSainsburys) [inline packages/nuget/Features.Tests/FeaturesClientTests.cs:34]: make test-feature a const - potentially in the scope of the class

File: packages/nuget/Features.Tests/Providers/NamespaceProviderTests.cs (hunk 1 above)
Comment (eboaden) [reply packages/nuget/Features.Tests/Providers/NamespaceProviderTests.cs]: This has been resolved

File: packages/nuget/Features.Tests/FeaturesClientTests.cs (hunk 2 above)
Comment (eboaden) [reply packages/nuget/Features.Tests/FeaturesClientTests.cs]: This has been resolved

File: packages/nuget/Features.Tests/FeaturesClientTests.cs (hunk 3 above)
Comment (eboaden) [reply packages/nuget/Features.Tests/FeaturesClientTests.cs]: This has been resolved

**Pull request merged**